compiler-rt/lib/scudo/standalone/secondary.h

```diff
@@ -122,7 +122,28 @@ bool mapSecondary(const Options &Options, uptr CommitBase, uptr CommitSize,
   Flags |= MAP_RESIZABLE;
   Flags |= MAP_ALLOWNOMEM;
 
-  const uptr MaxUnusedCacheBytes = MaxUnusedCachePages * getPageSizeCached();
+  const uptr PageSize = getPageSizeCached();
+  if (SCUDO_TRUSTY) {
+    /*
+     * On Trusty we need AllocPos to be usable for shared memory, which cannot
+     * cross multiple mappings. This means we need to split around AllocPos
+     * and not over it. We can only do this if the address is page-aligned.
+     */
+    const uptr TaggedSize = AllocPos - CommitBase;
```

Comment:
Per discussion, I think we may still want `TaggedSize != 0` or `DCHECK_NE(TaggedSize, 0u)`. Or we don't need them anymore?

Reply:
Maybe I misunderstood, wasn't the conclusion that `TaggedSize > 0` is always true? The only exception is when `mapSecondary` is called to free memory from `MapAllocatorCache::store`, but that code path could be disabled on Trusty instead.